File: app/src/main/java/com/permissionnanny/missioncontrol/SpacesItemDecoration.java
package com.permissionnanny.missioncontrol;

import android.content.Context;
import android.graphics.Rect;
import android.support.v7.widget.RecyclerView;
import android.util.DisplayMetrics;
import android.util.TypedValue;
import android.view.View;

public class SpacesItemDecoration extends RecyclerView.ItemDecoration {

    private int mPxLeft;
    private int mPxTop;
    private int mPxRight;
    private int mPxBottom;

    public SpacesItemDecoration(Context context, int unit, float left, float top, float right, float bottom) {
        DisplayMetrics dm = context.getResources().getDisplayMetrics();
        mPxLeft = (int) TypedValue.applyDimension(unit, left, dm);
        mPxTop = (int) TypedValue.applyDimension(unit, top, dm);
        mPxRight = (int) TypedValue.applyDimension(unit, right, dm);
        mPxBottom = (int) TypedValue.applyDimension(unit, bottom, dm);
    }

    @Override
    public void getItemOffsets(Rect outRect, View view, RecyclerView parent, RecyclerView.State state) {
        outRect.left = mPxLeft;
        outRect.top = mPxTop;
        outRect.right = mPxRight;
        outRect.bottom = mPxBottom;
    }
}

File: app/src/main/kotlin/com/permissionnanny/missioncontrol/SpacesItemDecoration.kt
package com.permissionnanny.missioncontrol

import android.content.Context
import android.graphics.Rect
import android.support.v7.widget.RecyclerView
import android.util.TypedValue
import android.view.View

class SpacesItemDecoration(
        context: Context,
        unit: Int,
        left: Float,
        top: Float,
        right: Float,
        bottom: Float)
    : RecyclerView.ItemDecoration() {

    private val pxLeft: Int
    private val pxTop: Int
    private val pxRight: Int
    private val pxBottom: Int

    init {
        val dm = context.resources.displayMetrics
        pxLeft = TypedValue.applyDimension(unit, left, dm).toInt()
        pxTop = TypedValue.applyDimension(unit, top, dm).toInt()
        pxRight = TypedValue.applyDimension(unit, right, dm).toInt()
        pxBottom = TypedValue.applyDimension(unit, bottom, dm).toInt()
    }

    override fun getItemOffsets(outRect: Rect, view: View, parent: RecyclerView, state: RecyclerView.State?) {
        outRect.left = pxLeft
        outRect.top = pxTop
        outRect.right = pxRight
        outRect.bottom = pxBottom
    }
}

